flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/TestFlinkUpsert.java

```diff
@@ -165,6 +165,52 @@ public void testUpsertAndQuery() {
     }
   }
 
+  @Test
+  public void testUpsertOptions() {
+    String tableName = "test_upsert_options";
+    LocalDate dt20220301 = LocalDate.of(2022, 3, 1);
+    LocalDate dt20220302 = LocalDate.of(2022, 3, 2);
+
+    tableUpsertProps.remove(TableProperties.UPSERT_ENABLED);
```

Comment:
we shouldn't modify the `tableUpsertProps`, as the side effect can affect other tests. It is better to construct a different map in this test method. I am a little surprised and confused why the CI tests passed.

Reply:
That makes sense. I changed it. :joy: